swim/test_utils.go

```diff
@@ -146,6 +146,52 @@ func bootstrapNodes(t *testing.T, testNodes ...*testNode) {
 		})
 		require.NoError(t, err, "node must bootstrap successfully")
 	}
+	return hostports
+}
+
+func waitForConvergence(t *testing.T, timeout time.Duration, testNodes ...*testNode) {
+	var wg sync.WaitGroup
+	for _, tn := range testNodes {
+		wg.Add(1)
+		// execute the Protocol period on all nodes until the dissemination list is exhausted
+		go func(tn *testNode) {
+			for len(tn.node.disseminator.changes) > 0 {
+				tn.node.gossip.ProtocolPeriod()
+			}
+			wg.Done()
+		}(tn)
+	}
+
+	finished := make(chan bool, 1)
```

Comment:
If we make this an unbuffered channel the routine that waits for the `WaitGroup` to finish could end up in a state where it could not produce to the channel if the timeout already occurred.